### PR TITLE
Invalid configuration handling

### PR DESCRIPTION
**Pull request by xXx-caillou-xXx**
As @buixor is suggesting in #225, add an option (disabled by default) to drop requests when the configuration is invalid.

### CODE REVIEW

File: src/snuffleupagus.c [hunk 1]
```diff
@@ -34,9 +34,21 @@ static inline void sp_op_array_handler(zend_op_array *op) {
 
 ZEND_DECLARE_MODULE_GLOBALS(snuffleupagus)
 
+static PHP_INI_MH(StrictMode) {
+  TSRMLS_FETCH();
+
+  SNUFFLEUPAGUS_G(is_strict_mode) = false;
```
Comment (jvoisin) [inline src/snuffleupagus.c]: Done your prefer `SNUFFLEUPAGUS_G(is_strict_mode) =  new_value && zend_string_equals_literal(new_value, "1");` instead?

Comment (xXx-caillou-xXx) [reply src/snuffleupagus.c]: sure

File: src/tests/broken_conf_strict_mode.phpt [hunk 2]
```diff
@@ -5,6 +5,9 @@ Broken configuration with strict mode enabled
 --INI--
 sp.configuration_file={PWD}/config/broken_conf.ini
 sp.strict_mode=On
+<?php
+echo 1337;
+?>
 --FILE--
```
Comment (xXx-caillou-xXx) [inline src/tests/broken_conf_strict_mode.phpt]: I think there is a problem here.

Comment (jvoisin) [reply src/tests/broken_conf_strict_mode.phpt]: How so?

Comment (xXx-caillou-xXx) [reply src/tests/broken_conf_strict_mode.phpt]: php code in the INI part

**Pull request merged**